### DISCUSSION TITLE
Delete folder recursively

### DISCUSSION THREAD

**Discussion by KamikaceDixit** [drakkan/sftpgo#1002] [Q&A] [answered]
Hello! I want to thank this incredible project.

I'm having a problem when I **try to delete a folder when files or other folders on it** in the website interface. I'm getting the message "Unable to delete directory "/Testfolder": failure". If I delete folder by folder from the deepest one, I can delete all of them, so it isn't a permission problem. **I think the interface it is not able to delete recursively**. I don't know if I'm doing something wrong or I missconfigured something.
Could you help me with this?

Thank you a lot!

**Comment by drakkan** [comment, marked as answer]
Hi,

this is not currently supported. The web UI simply reuses the same deletion method used for SFTP and thus mimics its behavior (deletion of non-empty folders is not allowed). Additionally, recursive deletion may take a long time for cloud backends with many nested files/folders

**Reply by KamikaceDixit** [reply]
I think it would be a great feature only in the web UI, as an option in the configuration. I understand it in the other aplications.

Thank you a lot for your time and contratulations for your great project!

**Reply by drakkan** [reply]
> I think it would be a great feature only in the web UI, as an option in the configuration. I understand it in the other aplications.
> 
> Thank you a lot for your time and contratulations for your great project!

to be honest I don't see real appreciation here. No sponsorship/donations, no GitHub star, just a new feature request you want added for free :disappointed:

**Comment by drakkan** [comment]
> Hello! I want to thank this incredible project.
> 
> I'm having a problem when I **try to delete a folder when files or other folders on it** in the website interface. I'm getting the message "Unable to delete directory "/Testfolder": failure". If I delete folder by folder from the deepest one, I can delete all of them, so it isn't a permission problem. **I think the interface it is not able to delete recursively**. I don't know if I'm doing something wrong or I missconfigured something. Could you help me with this?
> 
> Thank you a lot!

https://github.com/drakkan/sftpgo/commit/6f422c3d8b7f357ecbaa4075f5e43b1ade83bdce
